### ISSUE TITLE
timer.stop() is not working as expected

### ISSUE THREAD

**Issue by tourdefran** [SeverinDK/moment-timer#20]
Hi, great Moment plugin!
Sadly I'm having an issue, probably related to the stop() method. 

Sometimes it wont stop when the method is called and keeps looping, I'm using RequireJs and VelocityJs for some animations, I spent my last day debugging this, any help will be appreciated

I have uploaded [a sample to test it](http://ritariviera.com/fran/k8/last)
The play button starts the timer, and the stop should stop it, in the console I print the isStopped() status
sometimes brings false, and sometimes works,
thanks in advance,
Fran

**Comment by tourdefran** [comment]
Well, looks like this validation was the problem in my scenario: 
[https://github.com/SeverinDK/moment-timer/blob/master/lib/moment-timer.js#L44](https://github.com/SeverinDK/moment-timer/blob/master/lib/moment-timer.js#L44)

I just commented the lines 44-54 and now is working fine, what do you think, this can be a problem for other functionalities ( I'm not using the pause utility)?
Maybe a "resume" implementation is needed?

**Comment by SeverinDK** [comment]
Thank you for reporting this issue.

I have tested your code and I see what you mean. I will investigate tonight.

**Comment by SeverinDK** [comment]
I managed to locate and fix the issue. It will be published on NPM shortly.

The problem you experienced happened because `.duration()` was being called while the timer was running. Inside `.duration()` I would just clear the timer and then use `.stop()` and `.start()` to recreate the native JavaScript timer with the new duration, but this would cause issues because the `started` and `stopped` variables were then unexpectedly changed.

To overcome this issue I have added some new internal methods that will handle these things.

Let me know if you experience any other issues.
